sdk/keyvault/Azure.Security.KeyVault.Keys/src/JsonWebKey.cs

```diff
@@ -556,7 +556,7 @@ private void Initialize(ECDsa ecdsa, bool includePrivateParameters)
             KeyOps = new List<KeyOperation>(includePrivateParameters ? s_eCPrivateKeyOperation : s_eCPublicKeyOperation);
 
             ECParameters ecParameters = ecdsa.ExportParameters(includePrivateParameters);
-            CurveName = KeyCurveName.Find(ecParameters.Curve.Oid, ecdsa.KeySize).ToString() ?? throw new InvalidOperationException("elliptic curve name is invalid");
+            CurveName = KeyCurveName.FromOid(ecParameters.Curve.Oid, ecdsa.KeySize).ToString() ?? throw new InvalidOperationException("elliptic curve name is invalid");
```

Comment:
>CurveName = KeyCurveName.FromOid(ecParameters.Curve.Oid, ecdsa.KeySize).ToString() ?? throw new InvalidOperationException("elliptic curve name is invalid"); [](start = 12, length = 156)

Now that KeyCurveName is an extensible enum like struct we should change the property CurveName to be type KeyCurveName? (Nullable<KeyCurveName>) rather than string.

Reply:
Actually, if we go off the swagger it should be just "Curve" (idiomatically cased). This is something I was going to drive across the languages. However, swagger also suggests the type should be named JsonWebKeyCurveName, which we already knowingly violate. So I'm not sure how consistent we want to be across langauges. We should discuss offline.

Reply:
Also, in this case it wouldn't help to make it nullable since, to construct an `ECDsa` we need a supported curve name, which isn't reliable to look up by name (which is why I don't even use it) since different OSes use different names, in practice. But we could add something and just make sure devs understand it's all on them to get right.